x-pack/plugins/fleet/common/settings/types.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsFieldMapping } from '@kbn/core-saved-objects-server';
+import type { z } from 'zod';
+
+export interface SettingsConfig {
+  name: string;
+  title: string;
+  description: string;
+  learnMoreLink?: string;
+  schema: z.ZodTypeAny;
+  saved_object_field?: {
+    name: string;
+    mapping: SavedObjectsFieldMapping;
+  };
+  api_field: {
```

Comment:
used an object here so it easily extendable in the future, if we want to have some custom validation for api